packages/components/src/number-control/style.scss

```diff
@@ -0,0 +1,18 @@
+input.components-number-control__input {
+	display: inline-block;
+	font-weight: 500;
+	height: 30px;
+	width: 54px;
```

Comment:
Love the idea of a dedicated NumberControl but nervous about defining the width here. This makes it tougher to use as a full-width control. Maybe there could be a modifier class (`.components-number-control__input--narrow` or something like that) which applies the width constraint?